tests/src/ykcompile.rs

```diff
@@ -940,74 +940,74 @@ fn mut_lhs_struct_ref() {
 fn place_larger_than_reg() {
     #[derive(Debug, Eq, PartialEq)]
     struct S(u64, u64, u64);
-    struct IO(S);
+    struct InterpCtx(S);
 
     #[interp_step]
-    fn ten(io: &mut IO) {
+    fn ten(io: &mut InterpCtx) {
         io.0 = S(10, 10, 10);
     }
 
-    let mut inputs = IO(S(0, 0, 0));
+    let mut ctx = InterpCtx(S(0, 0, 0));
     let th = start_tracing(TracingKind::HardwareTracing);
-    ten(&mut inputs);
+    ten(&mut ctx);
     let sir_trace = th.stop_tracing().unwrap();
     let ct = compile_trace(sir_trace).unwrap();
-    assert_eq!(inputs.0, S(10, 10, 10));
+    assert_eq!(ctx.0, S(10, 10, 10));
 
-    let mut args = IO(S(1, 1, 1));
+    let mut args = InterpCtx(S(1, 1, 1));
     assert!(unsafe { ct.execute(&mut args).is_null() });
     assert_eq!(args.0, S(10, 10, 10));
 }
 
 #[ignore] // FIXME Sometimes assertion fails. Guard really failing?
 #[test]
 fn array_slice_index() {
-    struct IO<'a>(&'a [u8], u8);
+    struct InterpCtx<'a>(&'a [u8], u8);
 
     #[interp_step]
-    fn interp_step(io: &mut IO) {
+    fn interp_step(io: &mut InterpCtx) {
         io.1 = io.0[2];
     }
 
     let a = [1, 2, 3];
-    let mut inputs = IO(&a, 0);
+    let mut ctx = InterpCtx(&a, 0);
     let th = start_tracing(TracingKind::HardwareTracing);
-    interp_step(&mut inputs);
+    interp_step(&mut ctx);
     let sir_trace = th.stop_tracing().unwrap();
     let ct = compile_trace(sir_trace).unwrap();
-    let mut args = IO(&a, 0);
+    let mut args = InterpCtx(&a, 0);
     assert!(unsafe { ct.execute(&mut args).is_null() });
     assert_eq!(args.1, 3);
 }
 
 // Only `interp_step` annotated functions and their callees should remain after trace trimming.
 #[test]
 fn trim_junk() {
-    struct IO(u8);
+    struct InterpCtx(u8);
 
     #[interp_step]
-    fn interp_step(io: &mut IO) {
+    fn interp_step(io: &mut InterpCtx) {
         io.0 += 1;
     }
 
-    let mut inputs = IO(0);
+    let mut ctx = InterpCtx(0);
     let th = start_tracing(TracingKind::HardwareTracing);
-    interp_step(&mut inputs);
-    inputs.0 = 0; // Should get trimmed.
-    interp_step(&mut inputs);
-    inputs.0 = 0; // Should get trimmed
-    interp_step(&mut inputs);
+    interp_step(&mut ctx);
+    ctx.0 = 0; // Should get trimmed.
+    interp_step(&mut ctx);
+    ctx.0 = 0; // Should get trimmed
```

Comment:
Uh oh. "should". Is this OK in this context?

Reply:
This PR doesn't change that comment. It will need changing, but that's for a future PR.

Reply:
OK. But I feel that should is actually OK here.

To use an analogy: the test is like an exam question: the outcome is not fixed.

The following two statements are IMHO true:

 * You **should** answer the exam question correctly (it's in your interest) -- In the same way, you **should** ensure the test passes.
 * But you don't have to answer the exam question correctly -- The test **could** fail.
 * But to get a mark for this exam question, you **must** provide the correct answer -- To pass CI you **must** ensure the test passes.

And here starts my career as a philosopher...

Reply:
You're right but this isn't a change that should be done in this PR: it's not related to the PR's fundamental objective.

Reply:
What I'm saying is that no change is required. Thus no follow up PR. "should" is fine here IMHO.